### PR TITLE
First packaging of ag-grid, ag-grid-react and ag-grid-enterprise.

### PR DESCRIPTION
**Pull request by pupeno**
New package: **Extern:** Not needed as this is Closure library.

I think they are not necessary. I didn't define them and it works just fine.

### CODE REVIEW

**Comment by pupeno** [comment]
I'm not sure what failed here. What's the error?

**Comment by Deraen** [comment]
CI error looks like random error, or perhaps build is using more resources that CircleCI allows.

These are not currently packages as Closure libs, so externs are required. Did you try advanced compilation?

Closure libraries don't use `:foreign-libs` part of `deps.cljs` (which is generated by deps-cljs task), instead they use `:libs`: https://github.com/cljsjs/packages/blob/master/openlayers/resources/deps.cljs#L1 (and even real Closure libs might require externs).

Checking the original repo, I don't see any mention of Closure, and the source is using CommonJS modules instead of Closure modules.

**Comment by pupeno** [comment]
No, I didn't try :advanced. I'll do that.

**Comment by Deraen** [comment]
Ah no, there is a real problem with build:

```
                              java.lang.Thread.run                       Thread.java:  745
java.util.concurrent.ThreadPoolExecutor$Worker.run           ThreadPoolExecutor.java:  617
 java.util.concurrent.ThreadPoolExecutor.runWorker           ThreadPoolExecutor.java: 1142
               java.util.concurrent.FutureTask.run                   FutureTask.java:  266
                                               ...                                        
               clojure.core/binding-conveyor-fn/fn                          core.clj: 1938
                                 boot.core/boot/fn                          core.clj: 1029
                               boot.core/run-tasks                          core.clj: 1019
      cljsjs.boot-cljsjs.packaging/eval226/fn/G/fn                     packaging.clj:   86
      cljsjs.boot-cljsjs.packaging/eval63/fn/fn/fn                     packaging.clj:   25
     cljsjs.boot-cljsjs.packaging/eval114/fn/fn/fn                     packaging.clj:   40
                        boot.user/eval439/fn/fn/fn  boot.user2315155993321634664.clj:   29
                     boot.user/eval439/fn/fn/fn/fn  boot.user2315155993321634664.clj:   29
                                               ...                                        
                                      boot.util/sh                          util.clj:  397
                                clojure.core/apply                          core.clj:  646
                                               ...                                        
                    boot.from.me.raynes.conch/proc                         conch.clj:   40
                    java.lang.ProcessBuilder.start               ProcessBuilder.java: 1029
                       java.lang.ProcessImpl.start                  ProcessImpl.java:  134
                      java.lang.UNIXProcess.<init>                  UNIXProcess.java:  247
                 java.lang.UNIXProcess.forkAndExec                   UNIXProcess.java     
       java.io.IOException: error=2, No such file or directory
       java.io.IOException: Cannot run program "bower" (in directory "/home/ubuntu/.boot/cache/tmp/home/ubuntu/packages/ag-grid/gc8/2x1x9j/ag-grid-8.0.1"): error=2, No such file or directory
clojure.lang.ExceptionInfo: Cannot run program "bower" (in directory "/home/ubuntu/.boot/cache/tmp/home/ubuntu/packages/ag-grid/gc8/2x1x9j/ag-grid-8.0.1"): error=2, No such file or directory
    file: "/tmp/boot.user2315155993321634664.clj"
    line: 35
```

The error is in the middle of output, because the script tries to run all packages instead of stopping on the first that fails.

**Comment by pupeno** [comment]
Ah... got it. I thought that could be a problem. I'll fix that too.

**Comment by Deraen** [comment]
Externs only affect advanced compilation so using that is necessary to see if the externs are required.

**Comment by pupeno** [comment]
I added externs, one is generated and the other is manual.

File: ag-grid-enterprise/build.boot [hunk 1]
```diff
@@ -0,0 +1,61 @@
+(set-env!
+  :resource-paths #{"resources"}
+  :dependencies '[[cljsjs/boot-cljsjs "0.5.2" :scope "test"]])
```
Comment (Deraen) [inline ag-grid-enterprise/build.boot]: This package should depend on `cljsjs/ag-grid`, `:requires` on the `deps.cljs` alone doesn't add dependency on cljsjs package.

File: ag-grid-react/build.boot [hunk 2]
```diff
@@ -0,0 +1,67 @@
+(set-env!
+  :resource-paths #{"resources"}
+  :dependencies '[[cljsjs/boot-cljsjs "0.5.2" :scope "test"]
+                  [cljsjs/react "15.2.1-0" :scope "provided"]
+                  [cljsjs/react-dom "15.2.1-0" :scope "provided"]])
```
Comment (Deraen) [inline ag-grid-react/build.boot]: This package should depend on `cljsjs/ag-grid` probably?

File: ag-grid-react/build.boot [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+           (download-lib)
+           (build)
+           (sift :move {#".*ag-grid-react.inc.js" "cljsjs/ag-grid-react/development/ag-grid-react.inc.js"})
+           (sift :include #{#"^cljsjs"})
+           (minify :in "cljsjs/ag-grid-react/development/ag-grid-react.inc.js"
+                   :out "cljsjs/ag-grid-react/production/ag-grid-react.min.inc.js"
+                   :lang :ecmascript5)
+           (deps-cljs :name "cljsjs.ag-grid-react"
+                      :requires ["cljsjs.react"
+                                 "cljsjs.react.dom"])
```
Comment (Deraen) [inline ag-grid-react/build.boot]: This package probably depends on `cljsjs.ag-grid` foreign-lib?

**Comment by pupeno** [comment]
ag-grid-enterprise somehow hooks into ag-grid and enables new features without adding any API functions, just new options to ag-grid ones. I'm still trying to figure out how that would work. Can you approve it and we'll have patches soon as we are going to be using it right away at one of my client's.

File: ag-grid-react/build.boot (hunk 3 above)
Comment (pupeno) [inline ag-grid-react/build.boot]: Fixed.

File: ag-grid-react/build.boot (hunk 2 above)
Comment (pupeno) [inline ag-grid-react/build.boot]: Fixed.

File: ag-grid-enterprise/build.boot (hunk 1 above)
Comment (pupeno) [inline ag-grid-enterprise/build.boot]: Fixed.

**Pull request merged**